src/mongocxx/include/mongocxx/v1/client.hpp

```diff
@@ -35,12 +55,307 @@ namespace v1 {
 /// @attention This feature is experimental! It is not ready for use!
 ///
 class client {
+   private:
+    class impl;
+    void* _impl;
+
+   public:
+    class options;
+
+    ///
+    /// Destroy this object.
+    ///
+    /// @warning Invalidates all associated views, databases, cursors, client sessions, and change streams.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL() ~client();
+
+    ///
+    /// Move constructor.
+    ///
+    /// @par Postconditions:
+    /// - `other` is in an assign-or-destroy-only state.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL() client(client&& other) noexcept;
+
+    ///
+    /// Move assignment.
+    ///
+    /// @par Postconditions:
+    /// - `other` is in an assign-or-destroy-only state.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(client&) operator=(client&& other) noexcept;
+
+    ///
+    /// This class is not copyable.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL() client(client const& other);
+
+    ///
+    /// This class is not copyable.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(client&) operator=(client const& other);
+
+    ///
+    /// Initialize with the given URI.
+    ///
+    /// @par Postconditions:
+    /// - `*this` is connected to the MongoDB server(s).
+    ///
+    /// @throws mongocxx::v1::server_error when a server-side error is encountered and a raw server error is available.
```

Comment:
Remove? I do not expect a server error would be returned for the client constructor.
```suggestion
```